contracts/wallet/WalletFactory.sol

```diff
@@ -44,95 +44,49 @@ contract WalletFactory is Owned, Managed {
 
     event ModuleRegistryChanged(address addr);
     event ENSManagerChanged(address addr);
-    event GuardianStorageChanged(address addr);
     event WalletCreated(address indexed wallet, address indexed owner, address indexed guardian);
 
-    // *************** Modifiers *************************** //
-
-    /**
-     * @dev Throws if the guardian storage address is not set.
-     */
-    modifier guardianStorageDefined {
-        require(guardianStorage != address(0), "IGuardianStorage address not defined");
-        _;
-    }
-
     // *************** Constructor ********************** //
 
     /**
      * @dev Default constructor.
      */
-    constructor(address _moduleRegistry, address _walletImplementation, address _ensManager) public {
+    constructor(address _moduleRegistry, address _walletImplementation, address _ensManager, address _guardianStorage) public {
+        require(_moduleRegistry != address(0), "WF: ModuleRegistry address not defined");
+        require(_walletImplementation != address(0), "WF: WalletImplementation address not defined");
+        require(_ensManager != address(0), "WF: ENSManager address not defined");
+        require(_guardianStorage != address(0), "WF: GuardianStorage address not defined");
         moduleRegistry = _moduleRegistry;
         walletImplementation = _walletImplementation;
         ensManager = _ensManager;
+        guardianStorage = _guardianStorage;
     }
 
     // *************** External Functions ********************* //
-
     /**
      * @dev Lets the manager create a wallet for an owner account.
-     * The wallet is initialised with a list of modules and an ENS..
-     * The wallet is created using the CREATE opcode.
-     * @param _owner The account address.
-     * @param _modules The list of modules.
-     * @param _label ENS label of the new wallet, e.g. franck.
-     */
-    function createWallet(
-        address _owner,
-        address[] calldata _modules,
-        string calldata _label
-    )
-        external
-        onlyManager
-    {
-        _createWallet(_owner, _modules, _label, address(0));
-    }
-
-    /**
-     * @dev Lets the manager create a wallet for an owner account.
-     * The wallet is initialised with a list of modules, a first guardian, and an ENS..
+     * The wallet is initialised with a list of modules, a first guardian, and an ENS.
      * The wallet is created using the CREATE opcode.
      * @param _owner The account address.
      * @param _modules The list of modules.
      * @param _label ENS label of the new wallet, e.g. franck.
      * @param _guardian The guardian address.
      */
-    function createWalletWithGuardian(
+    function createWallet(
```

Comment:
Do we want to change the name of this method? I think it doesn't hurt to be explicit about our intentions but it also means less code change on the backend.

Reply:
I prefer the shorter, cleaner name. Otherwise it'll be more correct to call it `createWalletWithModulesLabelAndGuardian`. I have not yet understood the complexities of the code changes in the back but I don't think simplicity and clarity should be sacrificed for the sake of renaming two methods.